packages/neuron-ui/src/containers/Main/hooks.ts

```diff
@@ -27,34 +27,45 @@ const SYNC_INTERVAL_TIME = 4000
 const CONNECTING_BUFFER = 15_000
 let CONNECTING_DEADLINE = Date.now() + CONNECTING_BUFFER
 
+const isCurrentUrl = (url: string) => {
+  const id = currentNetworkIDCache.load()
+  const list = networksCache.load()
+  const cached = list.find(n => n.id === id)?.remote
+  return cached === url
+}
+
 export const useSyncChainData = ({ chainURL, dispatch }: { chainURL: string; dispatch: StateDispatch }) => {
   useEffect(() => {
     let timer: NodeJS.Timeout
     const syncBlockchainInfo = () => {
       Promise.all([getTipHeader(), getBlockchainInfo()])
         .then(([header, chainInfo]) => {
-          dispatch({
-            type: AppActions.UpdateChainInfo,
-            payload: {
-              tipBlockNumber: `${BigInt(header.number)}`,
-              tipBlockHash: header.hash,
-              tipBlockTimestamp: +header.timestamp,
-              chain: chainInfo.chain,
-              difficulty: BigInt(chainInfo.difficulty),
-              epoch: chainInfo.epoch,
-            },
-          })
+          if (isCurrentUrl(chainURL)) {
+            dispatch({
+              type: AppActions.UpdateChainInfo,
+              payload: {
+                tipBlockNumber: `${BigInt(header.number)}`,
+                tipBlockHash: header.hash,
+                tipBlockTimestamp: +header.timestamp,
```

Comment:
I am wondering why it is using the operation `+` here.

Reply:
The type of `header.timestamp` is string while that of `state.tipBlockTimestamp` is number.

Reply:
Ok, I see now. So this `+` operator is used to implicitly convert string to a number.
Maybe it is more readable by using `parseInt(header.timestamp)`.

Reply:
> Ok, I see now. So this `+` operator is used to implicitly convert string to a number.
> Maybe it is more readable by using `parseInt(header.timestamp)`.

`+` is better than `parseInt` in most cases cuz more types are correctly handled by the `+` operator, e.g. `null` is converted into `0` by `+null` while `NaN` by `parseInt(null)`

Reply:
Thanks for the clarification. I see it is intended for convenience. 

I think the point to use `parseInt` is that it can precisely return the value in context of number type. In other words, `NaN` returned by `parseInt` for a `null` value makes more sense in my view. 

So `parseInt(value) === NaN ? 0 : parseInt(value)` in here seems more readable and less potential security issues that would be introduced by the use of `+` operator, which basically coalesce certain values in an opinioned way.

Reply:
> Thanks for the clarification. I see it is intended for convenience.
> 
> I think the point to use `parseInt` is that it can precisely return the value in context of number type. In other words, `NaN` returned by `parseInt` for a `null` value makes more sense in my view.
> 
> So `parseInt(value) === NaN ? 0 : parseInt(value)` in here seems more readable and less potential security issues that would be introduced by the use of `+` operator, which basically coalesce certain values in an opinioned way.

In most cases I've met and read the `+` is preferable to the `parseInt`. BTW, the `parseInt(value) === NaN` is always `false`.

Reply:
I think the potential issue could be at the edge cases.

Try `isNaN(parseInt(null))`